src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingParameterResolver.cs

```diff
@@ -107,16 +108,8 @@ public override string Name
             public override string Resolve(string value)
             {
                 string format = GetFormatOrNull(value);
-
-                if (!string.IsNullOrEmpty(format))
-                {
-                    return DateTime.UtcNow.ToString(format, CultureInfo.InvariantCulture);
-                }
-                else
-                {
-                    // default to ISO 8601
-                    return DateTime.UtcNow.ToString("yyyy-MM-ddTHH-mm-ssK", CultureInfo.InvariantCulture);
-                }
+                var val = new SystemBindingData().UtcNow;
+                return BindingDataPathHelper.ConvertParameterValueToString(val, format);
```

Comment:
Although I like the feature and the goal, I'm not a fan of moving the logic out of the specialized resolver types and having a single type (in this case, the `SystemBindingData`) that is now responsible for knowing how to format a bunch of different values/types.

With this approach, we're also allocating more, causing additional casts and unnecessary boxing.